sdk/src/dataMapper/PortalDataMapper.ts

```diff
@@ -47,8 +47,46 @@ export default class PortalDataMapper extends BaseDataMapper<Portal> {
     return hash;
   }
 
-  async bulkAttest() {
-    throw new Error("Not implemented");
+  async simulateBulkAttest(
+    portalAddress: Address,
+    attestationPayloads: AttestationPayload[],
+    validationPayloads: string[][],
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const attestationPayloadsArg: any[] = [];
+    attestationPayloads.forEach(async (attestationPayload) => {
+      const matchingSchema = await this.veraxSdk.schema.findOneById(attestationPayload.schemaId);
+      const attestationData = encode(matchingSchema.schema, attestationPayload.attestationData);
+      attestationPayloadsArg.push([
+        attestationPayload.schemaId,
+        attestationPayload.expirationDate,
+        attestationPayload.subject,
+        attestationData,
+      ]);
+    });
+
+    try {
+      const { request } = await this.web3Client.simulateContract({
+        address: portalAddress,
+        abi: abiDefaultPortal,
+        functionName: "bulkAttest",
+        account: this.walletClient.account,
+        args: [attestationPayloadsArg, validationPayloads],
+      });
```

Comment:
It's not the first time that we write this code to simulate transactions. Maybe we can extract that into a private method taking the `functionName` and `arg `as parameters?  
But I'm not sure it would be relevant to create a more global method (passing the ABI and all).